packager/packager.cc

```diff
@@ -1282,9 +1314,40 @@ ast::ParsedFile validatePackage(core::Context ctx, ast::ParsedFile file) {
     if (!absPkg.exists()) {
         // We already produced an error on this package when producing its package info.
         // The correct course of action is to abort the transform.
+
         return file;
     }
 
+    auto &pkgInfo = PackageInfoImpl::from(absPkg);
+    for (auto &i : pkgInfo.importedPackageNames) {
+        auto &otherPkg = packageDB.getPackageInfo(i.name.mangledName);
+
+        // this might mean the other package doesn't exist, but that
+        // should have been caught already
+        if (!otherPkg.exists()) {
+            continue;
+        }
+
+        if (!otherPkg.visibleTo().empty()) {
+            bool allowed = false;
+            for (auto &other : otherPkg.visibleTo()) {
+                if (other == absPkg.fullName()) {
+                    allowed = true;
+                }
+            }
```

Comment:
```suggestion
            bool allowed = absl::c_any_of(otherPkg.visibleTo(), [&absPkg](const auto &other) { return other == absPkg.fullName(); });
```

...or something like that.